A39-xds-http-filters.md

```diff
@@ -0,0 +1,152 @@
+  same filter, which can be useful for cases like authorization policies.
+- [`typed_config`](https://github.com/envoyproxy/envoy/blob/2ee9543fbb0fccdb69d4dcd8cb57a19743afb94b/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L838):
+  This field contains the configuration message for the filter.  gRPC
+  will contain a registry of filter implementations, keyed by the proto
+  message type used for its configuration; thus, when gRPC sees an entry
+  with a given proto message type, it will know which filter to
+  instantiate.  Note that as a special case, if the proto message type is
+  `udpa.type.v1.TypedStruct`, then gRPC will look inside that type's
+  [`type_url`](https://github.com/cncf/udpa/blob/cc1b757b3eddccaaaf0743cbb107742bb7e3ee4f/udpa/type/v1/typed_struct.proto#L38)
+  field to determine the actual type; this allows for third-party
+  plugins to encode their configurations as a `google.protobuf.Struct`
+  proto instead of using a concrete proto message type that the client
+  may not be able to handle.
+
+When gRPC receives the `HttpConnectionManager` proto in the LDS
+response, it will validate the list of filters as follows:
+- Every filter in the list must be of a known type, as identified by the
+  `type_url` field.  Any unknown config type will cause the Listener
+  resource to be NACKed.
+- Every filter in the list must have a unique instance name.  Any
+  duplicate instance names will cause the Listener resource to be NACKed.
+
+Note that gRPC will support configuring multiple instances of the same
+filter implementation, each with its own name and config.
+
+In each of the `VirtualHost`, `Route`, and `WeightedCluster.ClusterWeight`
+protos, there is a field called `typed_per_filter_config`, which is a map
+from filter instance name to a corresponding filter-specific configuration
+proto message.  (Note that Envoy currently keys these maps by legacy
+filter names instead of by filter instance names, but this is expected
+to change, as per https://github.com/envoyproxy/envoy/issues/12274; gRPC
+will directly implement the desired semantics here.)
+
+When gRPC receives the `RouteConfiguration` proto, either in the LDS or
+RDS response, it will validate the contents of each
+`typed_per_filter_config` map as follows:
+- Every config in the list must be of a known type, as identified by the
+  map value proto message type.  Any unknown config type will cause the
+  resource to be NACKed.
+- Note that gRPC will *not* fail validation if the map key specifies a
+  filter instance name that does not exist in the `HttpConnectionManager`
+  filter list.  This is because during an update, the xDS client code
+  cannot know which `HttpConnectionManager` config is currently being
+  used.
+
+Unlike Envoy, which provides an API that filters must call to access their
+per-{VirtualHost,Route,ClusterWeight} config overrides, gRPC will
+require filters to construct a merged configuration, applying any
```

Comment:
In case of update of HttpFilters by the xDS server (e.g. a new filter being added) wouldn't the Envoy approach be better in terms of complexity and performance as compared to merged configuration?